### DISCUSSION TITLE
Setting up gateway problem (Lilygo)

### DISCUSSION THREAD

**Discussion by Olgmo** [PricelessToolkit/MailBoxGuard#27] [Q&A] [answered]
I have a LILYGO LORA32 T3_V1.6.1 (433MHZ) as recommended and where I am able to compile using ESP32 Dev Module as board and set "#define LORA_V1_6_OLED  1" in board.h
(I did a clean new download and started all over again to make sure i had not messed up something)

Compiling on Arduino 1.8.19 and uploading successful to the board.
But this does not run very well. 
When monitoring the serial it is looping in  reboot like this.
Here is the output:

`........⸮⸮⸮⸮L⸮)a11aմ⸮)m9&!m9J
Connecting to OGM
..
WiFi connected
IP address: 
192.168.1.155

assert failed: i2c_ll_set_bus_timing i2c_ll.h:109 (scl_high > 13)

Backtrace: 0x40082769:0x3ffb1ff0 0x4008db9d:0x3ffb2010 0x400930ca:0x3ffb2030 0x400e3da6:0x3ffb2160 0x400d8fcd:0x3ffb2180 0x400d3363:0x3ffb21b0 0x400d2805:0x3ffb21d0 0x400d3591:0x3ffb21f0 0x400d39c6:0x3ffb2210 0x400d2aa9:0x3ffb2230 0x400db478:0x3ffb2290

ELF file SHA256: 9586d3ec60cf206b

Rebooting...
0⸮⸮⸮%
%⸮⸮
j!⸮N⸮⸮⸮m⸮!⸮
Connecting to OGM
..
WiFi connected
IP address: 
192.168.1.155

assert failed: i2c_ll_set_bus_timing i2c_ll.h:109 (scl_high > 13)`

So it fails in "display.init();"

The only thing I've changed in the code is the frequency, Wifi and HA_IP
BTW I aslo tried the LoRa_Gateway_OLED.ino with same result.

Any suggestions?

**Comment by PricelessToolkit** [comment]
Hi and thanks for supporting my work! 

1. it's my fault that I didn't mention which board to choose for the gateway. I falsely assumed it was obvious.
2.  dont forget to enable "erase all flash before sketch upload"

![Arduino_gateway_TTGO](https://github.com/PricelessToolkit/MailBoxGuard/assets/81021972/c6fa06de-0348-4c53-a009-e22f8f21a6fe)

**Reply by Olgmo** [reply]
Thanks for your quick reply.
I should have mentioned all the combination I have tried out and your board setting suggested has also been tried without success.
When using the correct board setting I still have the "assert failed: i2c_ll_set_bus_timing i2c_ll.h:109 (scl_high > 13)" 

I know that the board is working on OpenMQTTGateway with display okay. (Though I have not compiled this image myself, so I dont know the display setting. It is flashed directly from browser)

To debug, I have added this to print 
Serial.println("Display init");
  display.init();
Serial.println("Init Okay");

and I dont get the "Init Okay" in monitor.

**Reply by PricelessToolkit** [reply]
Can you try to use these libraries https://github.com/PricelessToolkit/MailBoxGuard/tree/main/Code/Arduino/Arduino_libraries these errors are not related to hardware.

**Reply by Olgmo** [reply, marked as answer]
Sorry for all the trouble.
I finaly made it work. 
I removed ALL arduino and started over from clean installation.
Then I had to move all files under "Arduino_libraries" to "libraries". Then the compiled project was working.
I dont speak fluent Arduino so thats probably my issue :-)

**Comment by steuerzentrale** [comment]
I got the same error. I also used PricelessToolkits libraries and for the additional board I used ESP32 Version 2.0.14. With those libraries it worked fine, see more details here: https://github.com/PricelessToolkit/MailBoxGuard/discussions/26 I use the gateway with the Auto-MQTT sketch.
